common/lru_policy.go

```diff
@@ -0,0 +1,118 @@
+package common
+
+import (
+	"container/list"
+	"fmt"
+	"sync"
+)
+
+type CacheBlock struct {
+	sync.RWMutex
+	StartIndex int64
+	EndIndex   int64
+	Data       []byte
+	Last       bool //last block in the file?
+}
+
+//KeyPair: the list node containing both block key and cache block values
+type KeyPair struct {
+	key   int64
+	value *CacheBlock
+}
+
+//LRUCache definition for Least Recently Used Cache implementation
+type LRUCache struct {
+	sync.RWMutex
+	Capacity int64
+	List     *list.List              //DoublyLinkedList: node1->node2.... node:=KeyPair
+	Elements map[int64]*list.Element //blockKey:KeyPair
+	Occupied int64
+}
+
+//NewLRUCache: creates a new LRUCache object with the defined capacity
+func NewLRUCache(capacity int64) *LRUCache {
+	return &LRUCache{
+		Capacity: capacity,
+		List:     new(list.List),
+		Elements: make(map[int64]*list.Element),
+	}
+}
+
+//Get: returns the cache value stored for the key, cache hits the handle and moves the list pointer to front of the list
+func (cache *LRUCache) Get(bk int64) (*CacheBlock, bool) {
+	found := false
+	var cb *CacheBlock
+	if node, ok := cache.Elements[bk]; ok {
+		cb = getKeyPair(node).value
+		cache.List.MoveToFront(node)
+		found = true
+	}
+	return cb, found
+}
+
+//Put: Inserts the key,value pair in LRUCache.
+func (cache *LRUCache) Put(key int64, value *CacheBlock) {
+	if cache.Occupied >= cache.Capacity {
+		pair := getKeyPair(cache.List.Back())
+		cache.Remove(pair.key)
+	}
+	node := &list.Element{
+		Value: KeyPair{
+			key:   key,
+			value: value,
+		},
+	}
+	pointer := cache.List.PushFront(node)
+	cache.Occupied += (node.Value.(KeyPair).value.EndIndex - node.Value.(KeyPair).value.StartIndex)
+	cache.Elements[key] = pointer
+}
+
+func (cache *LRUCache) Print() {
+	for _, value := range cache.Elements {
+		fmt.Printf("Key:%+v,Value:%+v\n", getKeyPair(value).value.StartIndex, getKeyPair(value).value.EndIndex)
```

Comment:
convert this to log.debug

Reply:
issue with logger and importing it. Will keep as print for now as it is only used for debugging.

Reply:
whats the issue with importing it?